### ISSUE TITLE
How do I install python development tools on the latest VS2017RC？

### ISSUE THREAD

**Issue by reaCodes** [Microsoft/PTVS#2099]
Release Date: January 27, 2017 (build 26127.00)
-----------------------------
![abc](https://cloud.githubusercontent.com/assets/13817254/22393351/aab8f954-e53f-11e6-8bdc-16c149964b08.png)
---------------------------
URL: https://www.visualstudio.com/en-us/news/releasenotes/vs2017-relnotes

**Comment by andyp05** [comment]
Not happy they just deleted the feature without a big notice.

**Comment by zooba** [comment]
Yeah, we're real sorry about this (and it's obviously disappointing for us, too). Unfortunately there was no way for us to get a message out sooner, or to get an alternate installer ready in time :(

We're working on a way to install the Python and Data Science workloads on top of an existing VS install - so back to the old days (for now) - and that will show up here as soon as it's ready.

Until then, feel free to leave a comment here. That will subscribe you to the thread and you'll get an email when I post here that we're available again.

**Comment by andyp05** [comment]
Is there any way we can get the last release support (from early January) back in the meantime?

**Comment by reaCodes** [comment]
@zooba My Python program builds on VS2017RC, so I can not do without PTVS, please release your installation file as soon as possible, thanks.😁

**Comment by grenzi** [comment]
leaving a message here to subscribe. good luck!

**Comment by sharkbound** [comment]
This is a subscription comment, ignore me

**Comment by dammit20** [comment]
This is a subscription comment, ignore me

**Comment by irydlund** [comment]
This is a subscription comment, ignore me

**Comment by jed-frey** [comment]
How to subscribe to an issue without leaving a bunch of noise comments.

![2017-01-30-161742_1920x1080_scrot](https://cloud.githubusercontent.com/assets/223258/22442015/bcc6927c-e707-11e6-9cd3-6712664abeed.png)

**Comment by thedovahkiin909** [comment]
Jeez, does nobody see the huge subscribe button on the right?

On Jan 30, 2017 10:17 AM, "Isak Rydlund" <notifications@github.com> wrote:

This is a subscription comment, ignore me

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/Microsoft/PTVS/issues/2099#issuecomment-276089092>, or mute
the thread

**Comment by hickford** [comment]
While Visual Studio 2017 RC had Python 3.6 support, the most recent release of PTVS on GitHub does not https://github.com/Microsoft/PTVS/releases/tag/v2.2.6 

> Note that Python 3.6 is still not supported by this release

**Comment by zooba** [comment]
@jed-frey @thedovahkiin909 I was hoping to get a long list of comments to screenshot and use as evidence, which is why I didn't mention it :)

**Comment by zooba** [comment]
@hickford v2.2.6 is a maintenance release from our previous branch, so it has none of the work that went into VS 2017. We aren't planning to backport that work to VS 2015 (though it's probably not too difficult, tbh, but right now it's lower priority than supporting VS 2017).

**Comment by sanelm** [comment]
This is a subscription comment, ignore me (for screenshot purposes for zooba) :)

**Comment by zooba** [comment]
@sanelm I appreciate the comment, but now the cat's out of the bag I can't really get away with it ;)

It is helpful for us to have an idea of just how many people are really affected by this (like, badly enough to find our GitHub and complain), as we don't get as good an impression from download counts.

**Comment by andyp05** [comment]
Do you have an eta for an installer for vs 2017?
Just want to plan if I have to downgrade python to 3.5 and vs to 2015 to continue development.
Thanks

**Comment by tr00st** [comment]
Alternatively, are the build instructions ("Build Instructions for PTVS" page on the wiki) intended to be up to date? They point to "Python\Setup\BuildRelease.ps1" for building installers locally after discussing VS2017 building - which no longer appears to exist...

**Comment by zooba** [comment]
@andyp05 No ETA yet, I'm afraid. If you do downgrade to VS 2015, you can add Python 3.6 as a custom environment (and tell VS that it's 3.5) and it will 99% work, you just won't get editor support for some of the new syntax features.

**Comment by zooba** [comment]
@tr00st Yeah, the build instructions are out of date. The installers we used to have won't work well for VS 2017, and the current packages we build don't work at all unless you're part of the main VS installer.

Best option right now is to install the VS Extensibility workload for VS 2017, open our PythonTools.sln and build/run it. We don't have correct support for building valid VSIX files (working on it), and there are some incompatibilities since the last RC update that we'll be fixing (which means we will break code that currently works, and it will be broken until the next VS update comes out).

As you can see, it's quite a messy situation we're in right now - doing the best we can to get something together for people to use :)

**Comment by sharkbound** [comment]
@zooba you guys take your time, i know how long it can take to have to fix code ;). I would profer a complete mostly bug-free product over a hastily done buggy one.

~wishing u luck, Sharkbound

**Comment by zooba** [comment]
Just cross-referencing the other main report of this so we don't lose the link - https://developercommunity.visualstudio.com/content/problem/15097/python-support-in-visual-studio-2017-rc3-261270.html

**Comment by zooba** [comment]
@sharkbound Right now we have to prioritise installation and the quality related work (e.g. translate everything into 14 languages) that forced us to be taken out. If we don't already have an issue for f-strings (I think we do), feel free to file one, but it's not going to be high up the list for now :(

**Comment by jbhurst** [comment]
Subscribing.  Looking to test Python in 2017.  Having some problems getting project directory to refresh in 2015 that I hope will be fixed.

**Comment by asCii88** [comment]
This is not a subscription comment, don't ignore me. 

I'm frustrated and sickened after I spent almost a day on and off installing, updating and whatnot VS 2017 thinking I had some kind of cognitive problem because everybody on my team is working with this setup and I couldn't get Python Tools to even show up on the installer
I can't believe there is no real information apart from a  small mention in the third paragraph of a blog post (https://blogs.msdn.microsoft.com/visualstudio/2017/01/27/update-to-visual-studio-2017-release-candidate/) where the most thorough explaination is an answer buried in the comments from six hours ago that looks like crap from a marketing specialist. 

> It is helpful for us to have an idea of just how many people are really affected by this (like, badly enough to find our GitHub and complain), as we don't get as good an impression from download counts.

Our helplessness is what is helpful to you?! I can't believe I had to dig in the issues of your Github repositories to find an answer. This kind of corporate screw-you behavior is what turns developers away from Microsoft, for as much as open-sourcy you like to be lately.

**Comment by zooba** [comment]
@asCii88 Don't worry, we're not ignoring you. I'm sorry we weren't able to get a more obvious message out about this and that you spent far too much time trying to figure out what had happened. I also apologise for the lack of clarity in what I posted earlier and you quoted - given what has already happened, more feedback is helpful, but we don't deliberately manufacture stuff like this just to get a response.

It was a shock to our team as well (those of us working directly on Python Tools, though much of the larger Visual Studio team was very sympathetic) but the end result is going to be a product that is more usable by more people, and while the delay is unfortunate we will be back. Right now, we're still trying to figure out exactly how that will look, but as soon as we can we'll let everyone know.

**Comment by andyp05** [comment]
I think the issue here is that Microsoft ripped out a complete feature set with no notice that was working with no method of reinstalling the previous working version. This stopped development for many programmers willing to help Microsoft test a RC version of software. 

It has been two weeks and neither a fix or a link to reload the January version has been provided. I think the programming community understands bugs with RC version of software. This is different and inexcusable. I have moved to PyCharm reluctantly.

**Comment by InfernoCoder** [comment]
Take the time to add Python support. I would've used VS 2015 for a while, but the sad thing is I never could install it properly after my system was wiped :(

So far, the RC version is much, much, much more stable than all the VS 2015 RTMs. But please add Python support soon, IDLE is bad.

**Comment by KV1** [comment]
I am also stuck. My problem is that I am a faculty member and am in the middle of a course. I had developed all my python programs in the VS2015 environment, but got stuck after I upgraded to the 2017 edition. Please communicate as soon as the fix is done.

**Comment by CountBigBang** [comment]
I very much appreciate the team's effort to resolve this issue. But to all of us desperate for a working PTVS it would be REALLY helpful to get a rough estimate on a release date. Are we talking about hours, days or weeks before we can expect Python to be reintegrated into VS?

**Comment by zooba** [comment]
Thanks everyone for the support and understanding. It's still not clear to us how we're going to be releasing Python support, but the plan is definitely to have something when VS 2017 [releases next month](https://blogs.msdn.microsoft.com/visualstudio/2017/02/09/visual-studio-2017-launch-event-and-20th-anniversary/). Depending on how much work we have to do to get there, it might be one more preview before we are confident to call it stable, but hopefully if the installation works then everything after that will be unchanged from what we were ready to release (which was a much needed stability and reliability improvement over the last release we actually got to make).

So I guess the answer is weeks rather than hours, but it shouldn't take months.

**Comment by KV1** [comment]
Thanks. I look forward to the release. In the mean time I will manage with some other tools.

**Comment by jcsharp** [comment]
I find it quite unsettling to find out that the data science and python tools got suddenly pulled off from the release candidate. I hope to see them both back in VS2017 shortly.

**Comment by ryan413** [comment]
just wondering... is a more reliable intellisense something that is being looked at?

**Comment by jd-wheaton** [comment]
It would be excellent to get PTVS support back into 2017. I started a new project in 2017, to get an auto-update and the tooling removed. Back to VSC for now.

**Comment by denfromufa** [comment]
@zooba I would prefer better PTVS support in VS 2017 even if stand-alone, than rushing with unfinished product. Is VS 2017 C/C++ compiler good for Python 3.5/3.6 extensions or we should stay on VS 2015 for this?

**Comment by zooba** [comment]
@denfromufa I believe the new compiler (VC toolset v141) will be fine for Python 3.5 and 3.6 extensions, though I haven't had a chance to thoroughly test it yet. There is an option in VS 2017 to install and use the older toolset (VC toolset v140), so you can move to VS 2017 and keep using the older compiler if you like. (It's near the bottom of the list of options when you select the Desktop Development with C++ workload.)

**Comment by zooba** [comment]
I'm hoping to be able to make an official announcement some time this week, but it's currently looking like we'll have something available at the same time as VS 2017's release with an update planned for a couple of months later. The second update will be when we are officially stable.

Thanks everyone for your patience. We will end up with better tooling for Python, so while it's been painful and stressful for now it will be worth it in the end.

**Comment by zooba** [comment]
I've just posted an update at https://blogs.msdn.microsoft.com/pythonengineering/2017/02/22/python-in-vs2017/

Short summary:
* there'll be a VS 2017 preview installer at launch
* it will have all VS 2017 features plus Python support
* within a couple of months, we'll be back in the main installer

Thanks for sticking with us. We appreciate it!

**Comment by zooba** [comment]
To be completely clear, we won't have anything available before the March 7 launch date, though if you've already signed a Microsoft NDA​ (e.g. you're an MVP), send us an email at python@microsoft.com and we'll see if we can get you early access to help us test.

**Comment by sanelm** [comment]
Oh happy days! Thanks for listening to us :)

**Comment by denfromufa** [comment]
PTVS release on my birthday, so nice of you ;)

**Comment by KV1** [comment]
Great news. Thanks.

**Comment by BMarinov** [comment]
On one hand, sure, these are some great news. On the other hand, I dont know if anyone noticed that, but PTVS on VS2015 does not support Python 3.6. And as far as I know its a wontfix/support wont ever come. And this over 2 months after the release of 3.6. I am currently getting failed builds every time I try to start a project but the error window shows no issues whatsoever. The second attempt is always successful, but this wastes my time every time I make a more major change.

Today I finally found time to install VS2017 only to find out that Python support is removed completely, may or may not come at launch (no criticism to you) and im stuck with a barely working IDE for the next 10 days at least. I am really disappointed right now and I hope that whoever decided to pull PTVS out of the **release candidate** had a very good reason to do so.

**Comment by ejronin** [comment]
Well, 2017RC at least knows python is used with a plugin. On the other
hand, you could use an alternative IDE under a 30 day trial that lints 3.6.
There are at least three out there and their support for it is a bit
wanting.

I'm new to programming and chose Python as the language I wanted to learn
because of RPI and how approachable I found the material to learn. That
said, 3.6 has some cool things, I've opted for VS Code until VS 2017 is out
(but I do use 2017RC to get familiar with it as well as PyCharm EAP).
However, any time I mention that I code in Python and have 3.6 installed
I've been cautioned about using the 3.6 specific coding methods - rather
encouraged to stick to 3.5 and be mindful of some 2.7 methods because there
are many modules and useful tools that do not support 3.6 yet or ever. For
example I'm working with Kivy and Django. Django 1.10.5 only supports 3.5
and 1.11 will support 3.6 but it's only in dev. So, really... 3.6 new
features and items have almost no practical value as of yet. You could
still code with 3.5 methods as very few things seem to be depreciated in
the 3.6 environment.

**Comment by zooba** [comment]
Those who are eagerly awaiting VS 2017's release today, go via https://aka.ms/ptvs if you want the one with Python support. I'll have a blog post going out later in the morning with more details.

**Comment by denfromufa** [comment]
@zooba is there offline installer or iso for this? I tried creating one and the installer complains:

![image](https://cloud.githubusercontent.com/assets/7870949/23666430/9f2f12e0-0320-11e7-8f54-4f4b3be74e59.png)

**Comment by zooba** [comment]
@denfromufa [This page](https://docs.microsoft.com/en-us/visualstudio/install/use-command-line-parameters-to-install-visual-studio) has all the info, but I think you just want `--layout` rather than `/layout`.

**Comment by zooba** [comment]
@denfromufa Oh, and happy birthday ;)

**Comment by zooba** [comment]
And today's update is available at http://blogs.msdn.microsoft.com/pythonengineering/2017/03/07/python-support-in-vs2017/

**Comment by zooba** [comment]
You have no idea how pleased I am to be able to finally resolve this issue :)

**Comment by denfromufa** [comment]
@zooba thank you, happy birthday VS and `--layout` worked!

The full offline installer is 25GB, so be careful.

**Comment by ejronin** [comment]
This is a bittersweet announcement. While I’m pleased the teams are working to release a stable and full functioned  toolset, I also feel like a little kid waiting for Christmas where I know what I’m getting but it’s a matter of waiting….patiently…..staring at it….waiting. 

 

I’ll be glad when Python can be used in VS again. I use another IDE to also do work but I am most comfortable in what I’ve been able to do in VS. 

 

Patiently, 

 

Shawn

**Comment by cove9988** [comment]
it's frustrating. we need the Python back soon.....

**Comment by mos0711** [comment]
Is there any estimate (roughly) when the official support for Python is back?
It's kind of annoying that one is bound to develop (commercial and official) products using VS2015. VS2017 is so much smarter...

**Comment by czlowieku** [comment]
sub

**Comment by BrotLeib** [comment]
comeback would be great.

**Comment by hickford** [comment]
Good news! Python is back in Visual Studio 2017 as of version 15.2 (26430.04) released May 10, 2017 

https://www.visualstudio.com/en-us/news/releasenotes/vs2017-relnotes#python

See also https://blogs.msdn.microsoft.com/visualstudio/2017/05/12/a-lap-around-python-in-visual-studio-2017/ and https://www.visualstudio.com/vs/python
